From 01f67e438b4c73a45f5582f1f097a6f91ab360ca Mon Sep 17 00:00:00 2001
From: Martin Pitt <martinpitt@gnome.org>
Date: Wed, 27 Jun 2012 11:46:28 +0200
Subject: [PATCH 2/2] /mainloop/timeouts test: Reduce race condition

Due to load, particular traits of the architecture, or other circumstances, the
/mainloop/timeouts sometimes manages to call the "every
100 ms" timer loop only 9 times in 1050 ms.

This is an inherent race-condition in the test; allow it some slack and accept
9 times as well.

https://bugzilla.gnome.org/show_bug.cgi?id=678959
---
 glib/tests/mainloop.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/glib/tests/mainloop.c b/glib/tests/mainloop.c
index 3f0bcee..3cef639 100644
--- a/glib/tests/mainloop.c
+++ b/glib/tests/mainloop.c
@@ -179,7 +179,8 @@ test_timeouts (void)
 
   g_main_loop_run (loop);
 
-  g_assert_cmpint (a, ==, 10);
+  g_assert_cmpint (a, >=, 9);
+  g_assert_cmpint (a, <=, 10);
   g_assert_cmpint (b, ==, 4);
   g_assert_cmpint (c, ==, 3);
 
-- 
1.7.10.4

